pilot/pkg/proxy/envoy/v1/mixer.go

```diff
@@ -157,23 +201,26 @@ func mixerHTTPRouteConfig(mesh *meshconfig.MeshConfig, role model.Node, instance
 	}
 	v2 := &mccpb.HttpClientConfig{
 		MixerAttributes: &mpb.Attributes{
-			Attributes: map[string]*mpb.Attributes_AttributeValue{
-				AttrDestinationIP:  {Value: &mpb.Attributes_AttributeValue_BytesValue{net.ParseIP(role.IPAddress)}},
-				AttrDestinationUID: {Value: &mpb.Attributes_AttributeValue_StringValue{"kubernetes://" + role.ID}},
-			},
+			Attributes: map[string]*mpb.Attributes_AttributeValue{},
 		},
 		ServiceConfigs: map[string]*mccpb.ServiceConfig{},
 	}
 
+	var labels map[string]string
+	// Note: instances are all running on mode.Node named 'role'
+	// So instance labels are the workload / Node labels.
+	if len(instances) > 0 {
+		labels = instances[0].Labels
+	}
+	addStandardNodeAttributes(v2.MixerAttributes.Attributes, AttrDestinationPrefix, role, labels)
+
 	if role.Type == model.Sidecar && !outboundRoute {
 		// Don't forward mixer attributes to the app from inbound sidecar routes
 	} else {
 		v2.ForwardAttributes = &mpb.Attributes{
-			Attributes: map[string]*mpb.Attributes_AttributeValue{
-				AttrSourceIP:  {Value: &mpb.Attributes_AttributeValue_BytesValue{net.ParseIP(role.IPAddress)}},
-				AttrSourceUID: {Value: &mpb.Attributes_AttributeValue_StringValue{"kubernetes://" + role.ID}},
-			},
+			Attributes: map[string]*mpb.Attributes_AttributeValue{},
 		}
+		addStandardNodeAttributes(v2.ForwardAttributes.Attributes, AttrSourcePrefix, role, labels)
```

Comment:
Should outbound labels include the collection of labels from all service instances since, in the case of one proxy/workload with multiple services, we don't know which "source service" originates the request?

Reply:
Not sure. @rshriram, does the ServiceInstance's labels refer to the labels defined against the service or the underlying workload? For example, consider two k8s services with label selectors that select the same set of pods. Do the ServiceInstance's correspond to the label selectors or the superset of labels defined on the pod?

Reply:
The code uses the labels for the pod: https://github.com/istio/istio/blob/master/pilot/pkg/serviceregistry/kube/controller.go#L302